src/fal/planner/schedule.py

```diff
@@ -47,7 +47,9 @@ def create_group(
 
 @dataclass
 class Scheduler:
-    groups: list[TaskGroup]
+    groups: List[TaskGroup]
+    failed_groups: List[TaskGroup] = field(default_factory=list)
+    skipped_groups: List[TaskGroup] = field(default_factory=list)
```

Comment:
Maybe we could add our version of `NodeStatus` and keep everything in a list. Like having an enum for `PENDING`, `SUCCEED`, `FAILED`, `SKIPPED`. Don't know though, this version seems very simple too. Any suggestions?

Reply:
> Maybe we could add our version of `NodeStatus` and keep everything in a list. Like having an enum for `PENDING`, `SUCCEED`, `FAILED`, `SKIPPED`

I would prefer this, tbh. then getting `failed_groups`/`skipped_groups` can be a function that just filters. But I am also OK with whatever you prefer.

Reply:
Are we going to change this?